File: api-go/lib/password.go
package lib

import "golang.org/x/crypto/bcrypt"

func HashPassword(password string) (string, error) {
	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
	if err != nil {
		return "", err
	}
	return string(hashedPassword), nil
}

// Returns nil on success, or an error on failure
func VerifyPassword(hash, password string) error {
	return bcrypt.CompareHashAndPassword([]byte(hash), []byte(password))
}

File: api-node/src/lib/password.ts
import { bcrypt, bcryptVerify } from 'hash-wasm'

const salt = new Uint8Array(16)
const costFactor = 10

export const hashPassword = async (password: string) =>
    bcrypt({ password, salt, costFactor })

export const verifyPassword = async (hash: string, password: string) =>
    bcryptVerify({ password, hash })
